File: file-upload/file-upload-scala-sbt/fileupload-impl/src/main/scala/com/example/fileupload/impl/FileuploadServiceImpl.scala
package com.example.fileupload.impl

import com.example.fileupload.api.FileUploadService
import com.lightbend.lagom.scaladsl.api.ServiceCall

import scala.concurrent.Future

class FileUploadServiceImpl extends FileUploadService {

  /**
    * This service call is implemented to have a basic Lagom ServiceImpl that we
    * can merge with the Play Router in FileUploadLoader.
    */
  override def uppercaseEcho() = ServiceCall { input =>
    Future.successful(input.toUpperCase)
  }

}

File: file-upload/file-upload-java-sbt/fileupload-impl/src/main/java/com/example/fileupload/impl/FileUploadServiceImpl.java
package com.example.fileupload.impl;

import com.example.fileupload.api.FileUploadService;
import com.lightbend.lagom.javadsl.api.ServiceCall;

import static java.util.concurrent.CompletableFuture.completedFuture;

public class FileUploadServiceImpl implements FileUploadService {

    @Override
    public ServiceCall<String, String> uppercaseEcho() {
        return input -> completedFuture(input.toUpperCase());
    }

}
